### PR TITLE
CQL: Fix supported options

### PR DESCRIPTION
**Pull request by mpenick**
Fixes https://github.com/stargate/stargate/issues/376

### CODE REVIEW

**Review by dimas-b — APPROVED** [review]

File: testing/src/test/java/io/stargate/it/cql/SupportedOptionsTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class SupportedOptionsTest extends BaseOsgiIntegrationTest {
+  /** Custom session builder that records supported options. */
+  private static class RecordingSupportedOptionsSessionBuilder
```
Comment (olim7t) [inline testing/src/test/java/io/stargate/it/cql/SupportedOptionsTest.java:50]: You can access the options using the driver's internal APIs:
```java
import com.datastax.oss.driver.internal.core.channel.DriverChannel;
import com.datastax.oss.driver.internal.core.context.InternalDriverContext;

private static Map<String, List<String>> getSupportedOptions(CqlSession session) {
  DriverChannel controlChannel =
      ((InternalDriverContext) session.getContext()).getControlConnection().channel();
  assertThat(controlChannel).as("Expected control connection to be connected").isNotNull();
  return controlChannel.getOptions();
}
```
This should allow you to write this test with the regular `CqlSessionExtension`.

**Review by olim7t — CHANGES_REQUESTED** [review]
LGTM, but a suggestion to simplify the test.

File: testing/src/test/java/io/stargate/it/cql/SupportedOptionsTest.java (hunk 1 above)
Comment (mpenick) [reply testing/src/test/java/io/stargate/it/cql/SupportedOptionsTest.java:50]: Thanks.

**Comment by mpenick** [comment]
Fixes https://github.com/stargate/stargate/issues/376

**Review by olim7t — APPROVED** [review]

**Pull request merged**